nlmod/dims/time.py

```diff
@@ -244,9 +307,138 @@ def set_ds_time(
         logger.error(msg)
         raise ValueError(msg)
 
+    # create time coordinates
     ds = ds.assign_coords(coords={"time": time})
     ds.coords["time"].attrs = dim_attrs["time"]
 
+    # add steady, nstp and tsmult to dataset
+    ds = set_time_variables(
+        ds, start, time, steady, steady_start, time_units, nstp, tsmult
+    )
+
+    return ds
+
+
+def set_ds_time_numerical(
```

Comment:
very nitpicky, but set_ds_time_numeric() might be grammatically slightly better?